packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,370 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { IResource, Names, Resource, Tag } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage, TagOption } from './constants';
+import { AddEventNotificationsProps, SetLaunchRoleProps, AddProvisioningRulesProps, AllowTagUpdatesProps, StackSetConstraintProps } from './constraints';
+import { AssociationManager } from './private/association-manager';
+import { getIdentifier } from './private/util';
+import { InputValidator } from './private/validation';
+import { IProduct } from './product';
+import { CfnPortfolio, CfnPortfolioPrincipalAssociation, CfnPortfolioShare } from './servicecatalog.generated';
+
+
+/**
+ * A Service Catalog portfolio.
+ */
+export interface IPortfolio extends IResource {
+
+  /**
+   * The ARN of the portfolio.
+   *
+   * @attribute
+   */
+  readonly portfolioArn: string;
+
+  /**
+   * The ID of the portfolio.
+   *
+   * @attribute
+   */
+  readonly portfolioId: string;
+
+  /**
+   * The name of the portfolio.
+   *
+   * @attribute
+   */
+  readonly portfolioName: string;
+
+  /**
+   * Associate portfolio with a principal
+   * (Role/Group/User).
+   * @param principal an IAM principal role
+   */
+  giveAccess(principal: iam.IIdentity): void;
+
+  /**
+   * Associate portfolio with the given product.
+   * @param product A service catalog produt.
+   */
+  addProduct(product: IProduct): void;
+
+  /**
+   * Share portfolio with another account.
+   * @param accountId AWS account to share portfolio with
+   * @param shareTagOptions ability to share or not share tag options
+   */
+  share(accountId: string, shareTagOptions?: boolean, acceptLanguage?: AcceptLanguage): void;
+
+  /**
+   * Add a Launch Notification Constraint.
+   */
+  addEventNotifications(props: AddEventNotificationsProps): void;
```

Comment:
Yea based on the guidance of the Design Guidelines:

>our APIs are based on the mental model of the user, and not the mental model of the service APIs

I felt like Constraints could be construed as the model of the service API and not necessarily the user, but I am open to this naming suggestion if you think this doesn't violate that guideline

Reply:
In my opinion, the mental model here is "I'm adding a constraint". If that's the case, I think the only name that makes sense is `addLaunchNotificationConstraint()`.

Reply:
To just follow up on, "constraints" aren't even a good way to describe all of these items, it was legacy terminology that was applied to anything that was an association type.  We want to experiment and get user feedback if these truer conventions make sense, because users get confused when a constraint doesn't actually constrain anything for example